### ISSUE TITLE
Suggestion: Rework `IPv6.prototype.toString` to condense the maximum number of `0` parts

### ISSUE THREAD

**Issue by JamesMGreene** [whitequark/ipaddr.js#85]
Suggestion: Rework `IPv6.prototype.toString` to condense the maximum number of `0` parts

While using this module, I've noticed that the IPv6 `toString` method does its part-condensing at the earliest valid point (where 2+ consecutive parts are `0`) in the sequence.

While that behavior is completely valid by IPv6 specification standards, I think it would be a nice optimization to condense the _maximum_ number of `0` parts within the sequence instead to potentially produce smaller strings.  If there are multiple spots with the same number of parts to condense, then it can prefer the first spot it encounters.

Example:

```js
// Condense the maximum consecutive number of `0` parts rather than the first
var ipv6String = ipaddr.parse( '0:0:ff:0:0:0:ab:cd' ).toString();

/*
Optimization:    "0:0:ff::ab:cd"
Actual/Current:  "::ff:0:0:0:ab:cd"
*/


// If there are multiple spots in the sequence that contain the same maximum number of `0` parts,
// then just condense the first like the module currently does today
var ipv6StringMultiples = ipaddr.parse( '0:0:ff:0:0:ab:0:0' ).toString();

/*
Optimization:    "::ff:0:0:ab:0:0"
Actual/Current:  "::ff:0:0:ab:0:0"
*/

var ipv6StringExtreme = ipaddr.parse( '0:0:ff::' ).toString();

/*
Optimization:    "0:0:ff::"
Actual/Current:  "::ff:0:0:0:0:0:0"
*/
```

**Comment by silverwind** [comment]
Sounds good to me, smaller is always better when it comes to IPv6 :)

**Comment by whitequark** [comment]
Is there some real-world case where this matters? Anyway, sure, why not.

**Comment by sarus** [comment]
This would be nice because it would bring ipaddr into compliance with RFC 5952.  This RFC is being followed by other libraries such as the PostgreSQL ip4r module as well as other IP libraries such as those for Qt and .Net.  This just makes it easier to ensure a consistent format is being used across a platform that might have multiple technologies involved.  

Here is a link to the RFC: https://tools.ietf.org/html/rfc5952

Here is the relevant section:

```
4.2.3.  Choice in Placement of "::"

   When there is an alternative choice in the placement of a "::", the
   longest run of consecutive 16-bit 0 fields MUST be shortened (i.e.,
   the sequence with three consecutive zero fields is shortened in 2001:
   0:0:1:0:0:0:1).  When the length of the consecutive 16-bit 0 fields
   are equal (i.e., 2001:db8:0:0:1:0:0:1), the first sequence of zero
   bits MUST be shortened.  For example, 2001:db8::1:0:0:1 is correct
   representation.
```

**Comment by whitequark** [comment]
I see. PRs welcome.

**Comment by cableray** [comment]
Implemented this behavior in this pr: #116 .  In order to maintain backwards compliance, that PR introduces a new method, but does compress the largest sequence of 0s, or the first in the case of ties.

**Comment by whitequark** [comment]
This is fixed now.
